### PR TITLE
Whatsnew1.3.0 links

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Part of #41747

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -247,7 +247,7 @@ These are bug fixes that might have notable behavior changes.
 ``Categorical.unique`` now always maintains same dtype as original
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Previously, when calling :meth:`~Categorical.unique` with categorical data, unused categories in the new array
+Previously, when calling :meth:`Categorical.unique` with categorical data, unused categories in the new array
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:250]: this link broken anyway

File: doc/source/whatsnew/v1.3.0.rst [hunk 2]
```diff
@@ -659,29 +659,29 @@ Deprecations
 - Deprecated allowing scalars to be passed to the :class:`Categorical` constructor (:issue:`38433`)
 - Deprecated constructing :class:`CategoricalIndex` without passing list-like data (:issue:`38944`)
 - Deprecated allowing subclass-specific keyword arguments in the :class:`Index` constructor, use the specific subclass directly instead (:issue:`14093`, :issue:`21311`, :issue:`22315`, :issue:`26974`)
-- Deprecated ``astype`` of datetimelike (``timedelta64[ns]``, ``datetime64[ns]``, ``Datetime64TZDtype``, ``PeriodDtype``) to integer dtypes, use ``values.view(...)`` instead (:issue:`38544`)
+- Deprecated :meth:`astype` of datetimelike (``timedelta64[ns]``, ``datetime64[ns]``, ``Datetime64TZDtype``, ``PeriodDtype``) to integer dtypes, use ``values.view(...)`` instead (:issue:`38544`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:662]: astype broken link

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @rhshadrach

some other broken links:

Styler.to_html on L145 is broken.
MultiIndex.set_names and MultiIndex.rename
 Window
Series.__getitem__() and DataFrame.__getitem__()
Styler.hidden_index and Styler.hidden_columns
Styler
Series.loc.__getitem__() and Series.loc.__setitem__()
FloatingArray 
Rolling, Expanding, and ExponentialMovingWindow
Categorical.unique()
GroupBy.rolling()
astype()
MultiIndex.is_lexsorted() and MultiIndex.lexsort_depth(), use MultiIndex.is_monotonic_increasing()
Rolling.win_type
Rolling.is_datetimelike
ExponentialMovingWindow.vol()
DatetimeIndex.union()
Categorical._codes
MultiIndex.drop_duplicates

and many others. not sure is all can be fixed. maybe rewording is required in some cases.

**Comment by simonjayhawkins** [comment]
> some other broken links:

another pass will be fine, don't need to add to this PR. if you can merge master to resolve conflicts

File: doc/source/whatsnew/v1.3.0.rst [hunk 3]
```diff
@@ -277,10 +277,10 @@ As an example of this, given:
         unique
         original.dtype == unique.dtype
 
-Preserve dtypes in  :meth:`~pandas.DataFrame.combine_first`
+Preserve dtypes in  :meth:`DataFrame.combine_first`
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:280]: doesn't this warn (line lenght)?

File: doc/source/whatsnew/v1.3.0.rst [hunk 4]
```diff
@@ -1011,66 +1011,66 @@ Period
...
 - Bug in :meth:`DataFrame.rolling` returning mean zero for all ``NaN`` window with ``min_periods=0`` if calculation is not numerical stable (:issue:`41053`)
 - Bug in :meth:`DataFrame.rolling` returning sum not zero for all ``NaN`` window with ``min_periods=0`` if calculation is not numerical stable (:issue:`41053`)
-- Bug in :meth:`SeriesGroupBy.agg` failing to retain ordered :class:`CategoricalDtype` on order-preserving aggregations (:issue:`41147`)
-- Bug in :meth:`DataFrameGroupBy.min` and :meth:`DataFrameGroupBy.max` with multiple object-dtype columns and ``numeric_only=False`` incorrectly raising ``ValueError`` (:issue:41111`)
-- Bug in :meth:`DataFrameGroupBy.rank` with the GroupBy object's ``axis=0`` and the ``rank`` method's keyword ``axis=1`` (:issue:`41320`)
+- Bug in :meth:`.SeriesGroupBy.agg` failing to retain ordered :class:`CategoricalDtype` on order-preserving aggregations (:issue:`41147`)
+- Bug in :meth:`.GroupBy.min` and :meth:`.GroupBy.max` with multiple object-dtype columns and ``numeric_only=False`` incorrectly raising ``ValueError`` (:issue:41111`)
```
[the remark this answers is not in the document]
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:1058]: Correct.

File: doc/source/whatsnew/v1.3.0.rst [hunk 5]
```diff
@@ -220,12 +220,12 @@ Other enhancements
...
-- :meth:`pandas.read_stata` and :class:`StataReader` support reading data from compressed files.
-- Add support for parsing ``ISO 8601``-like timestamps with negative signs to :meth:`pandas.Timedelta` (:issue:`37172`)
+- :func:`read_stata` now supports reading data from compressed files (:issue:`26599`)
+- Add support for parsing ``ISO 8601``-like timestamps with negative signs to :class:`Timedelta` (:issue:`37172`)
 - Add support for unary operators in :class:`FloatingArray` (:issue:`38749`)
 - :class:`RangeIndex` can now be constructed by passing a ``range`` object directly e.g. ``pd.RangeIndex(range(3))`` (:issue:`12067`)
-- :meth:`round` being enabled for the nullable integer and floating dtypes (:issue:`38844`)
-- :meth:`pandas.read_csv` and :meth:`pandas.read_json` expose the argument ``encoding_errors`` to control how encoding errors are handled (:issue:`39450`)
+- :meth:`Series.round` and :meth:`DataFrame.round` now work with nullable integer and floating dtypes (:issue:`38844`)
+- :meth:`read_csv` and :meth:`read_json` expose the argument ``encoding_errors`` to control how encoding errors are handled (:issue:`39450`)
```
[the remark this answers is not in the document]
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:228]: Yep - thanks (though I don't think it changes the resulting page, certainly +1 on consistency here)

File: doc/source/whatsnew/v1.3.0.rst (hunk 2 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:662]: Agreed - but no clear way to me to fix. Any suggestion?

File: doc/source/whatsnew/v1.3.0.rst (hunk 3 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:280]: Thanks for catching this - I rebuilt the docs locally and got no warnings. CI is showing the same - not sure why that is.

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:250]: Right, but the difference is that ``:meth:`~Categorical.unique`` just shows "unique" whereas the change has it show up as "Categorical.unique". I think the latter is preferred.

File: doc/source/whatsnew/v1.3.0.rst [hunk 6]
```diff
@@ -1098,36 +1098,36 @@ ExtensionArray
...
+- Bug in :func:`.infer_dtype` not recognizing Series, Index or array with a period dtype (:issue:`23553`)
+- Bug in :func:`.infer_dtype` raising an error for general :class:`.ExtensionArray` objects. It will now return ``"unknown-array"`` instead of raising (:issue:`37367`)
 - Bug in constructing a :class:`Series` from a list and a :class:`PandasDtype` (:issue:`39357`)
 - ``inspect.getmembers(Series)`` no longer raises an ``AbstractMethodError`` (:issue:`38782`)
 - Bug in :meth:`Series.where` with numeric dtype and ``other = None`` not casting to ``nan`` (:issue:`39761`)
 - :meth:`Index.where` behavior now mirrors :meth:`Index.putmask` behavior, i.e. ``index.where(mask, other)`` matches ``index.putmask(~mask, other)`` (:issue:`39412`)
 - Bug in :func:`pandas.testing.assert_series_equal`, :func:`pandas.testing.assert_frame_equal`, :func:`pandas.testing.assert_index_equal` and :func:`pandas.testing.assert_extension_array_equal` incorrectly raising when an attribute has an unrecognized NA type (:issue:`39461`)
```
[the remark this answers is not in the document]
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:1127]: Thanks!

**Comment by rhshadrach** [comment]
> some other broken links:
> ...
> and many others. not sure is all can be fixed. maybe rewording is required in some cases.

Thanks @simonjayhawkins - I've gone through your list and fixed the `Styler` and `Styler.to_html`.

For classes where we'd like to link to but don't have docs (e.g. attributes of a MultiIndex), we could certainly reword. For example:

> - Add support for dict-like names in :meth:`MultiIndex.set_names` and :meth:`MultiIndex.rename` (:issue:`20421`)

This could be reworded as

> - Add support for dict-like names in :meth:`Index.set_names` and :meth:`Index.rename` in the case of a :class:`Multiindex` (:issue:`20421`)

My opinion is that it'd be better to have docs for `MultiIndex.set_names` rather than change the phrasing of the whatsnew (but perhaps that poses its own challenges?). Even if consensus is to change the wording instead, I'd still like to consider such modifications out of scope for this PR.

For dunder methods, when there is a clear fix I plan to make them in a subsequent PR, however many cases it isn't clear to me how the docs should be changed.

File: doc/source/whatsnew/v1.3.0.rst (hunk 3 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst]: From reST docs (https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections):

> An underline/overline is a single repeated punctuation character that begins in column 1 and forms a line extending at least as far as the right edge of the title text. Specifically, an underline/overline character may be any non-alphanumeric printable 7-bit ASCII character 5. When an overline is used, the length and character used must match the underline.

**Comment by simonjayhawkins** [comment]
Thanks @rhshadrach for the updates.


merging as is, so that can continue with #41747 and since merge conflicts are likely to quickly arise with this .

comments here can be addressed in another pass.

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.3.0.rst]: didn't mean for this to be changed. feel free to change this again. but was thinking that could maybe reword instead.

File: doc/source/whatsnew/v1.3.0.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.3.0.rst:228]: yes. `:meth:` and `:func:` seem interchangeable. not sure why we have both.

File: doc/source/whatsnew/v1.3.0.rst (hunk 3 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.3.0.rst]: I think, we had some cases where I forgot to reduce the underline when changing release dates in previous whatsnew. didn't seem to do any harm, and rendered fine.

File: doc/source/whatsnew/v1.3.0.rst (hunk 1 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst]: Doh! I changed it just to make sure my understanding of the behavior was correct, but forgot to change it back. The fix is now in #41822
